delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -0,0 +1,469 @@
+            if query_para is None or \
+                    (query_para['begin_time']
+                     <= occur_time
+                     <= query_para['end_time']):
+                alert_model = {
+                    'alert_id': alert_map['AlertID'],
+                    'alert_name': alert_map['ProbableCause'],
+                    'severity': constant.ALERT_SEVERITY
+                    [alert_map['PerceivedSeverity']],
+                    'category': constants.Category.FAULT,
+                    'type': constants.EventType.EQUIPMENT_ALARM,
+                    'occur_time': occur_time,
+                    'description': alert_map['Description'],
+                    'match_key': hashlib.md5(
+                        (alert_map['AlertID'] +
+                         str(occur_time)).encode()).hexdigest(),
+                    'resource_type': constants.DEFAULT_RESOURCE_TYPE,
+                    'location': alert_map['AlertingResourceName']
+                }
+                alert_list.append(alert_model)
+        return alert_list
+
+    def list_alerts(self, query_para):
+        try:
+            alert_list = []
+            """Query the two alarms separately"""
+            alert_list += self.get_events(query_para)
+            alert_list += self.get_alerts(query_para)
+            return alert_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage alert from " \
+                      "netapp cmode: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage alert from " \
+                      "netapp cmode: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def clear_alert(self, alert):
+        try:
+            ssh_command = \
+                constant.CLEAR_ALERT_COMMAND + alert['alert_id']
+            self.ssh_pool.do_exec(ssh_command)
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage alert from " \
+                      "netapp cmode: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage alert from " \
+                      "netapp cmode: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_disks(self, storage_id):
+        disks_list = []
+        physicals_list = []
+        disks_info = self.ssh_pool.do_exec(
+            constant.DISK_SHOW_DETAIL_COMMAND)
+        disks_array = disks_info.split(
+            constant.DISK_SPLIT_STR)
+        physicals_info = self.ssh_pool.do_exec(
+            constant.DISK_SHOW_PHYSICAL_COMMAND)
+        disks_map = {}
+        physical_array = physicals_info.split('\r\n')
+        speed = physical_type = firmware = '-'
```

Comment:
Please make sure whether these default value is ok

Reply:
Has been modified